maestro-driver-ios/maestro-driver-iosUITests/Routes/Handlers/SwipeRouteHandler.swift

```diff
@@ -5,63 +5,40 @@ import os
 class SwipeRouteHandler: RouteHandler {
     
     private let logger = Logger(subsystem: Bundle.main.bundleIdentifier!, category: "SwipeRouteHandler")
-    
+
     func handle(request: FlyingFox.HTTPRequest) async throws -> FlyingFox.HTTPResponse {
-        guard let appId = request.query["appId"] else {
-            logger.error("Requested view hierarchy for an invalid appId")
-            return HTTPResponse(statusCode: HTTPStatusCode.badRequest)
-        }
-        
         let decoder = JSONDecoder()
         
         guard let requestBody = try? decoder.decode(SwipeRequest.self, from: request.body) else {
             let errorData = handleError(message: "incorrect request body provided")
             return HTTPResponse(statusCode: HTTPStatusCode.badRequest, body: errorData)
         }
-        
-        let response = await MainActor.run {
-            let xcuiApplication = XCUIApplication(bundleIdentifier: appId)
-            
-            let element = xcuiApplication
-            
-            let velocity: XCUIGestureVelocity
-            if let v = requestBody.velocity {
-                velocity = XCUIGestureVelocity(CGFloat(v))
-            } else {
-                velocity = XCUIGestureVelocity.default
-            }
-            
-            let startPoint = element.coordinate(withNormalizedOffset: CGVector(
-                    dx: CGFloat(requestBody.startX),
-                    dy: CGFloat(requestBody.startY)
-            ))
-            
-            let endPoint = element.coordinate(withNormalizedOffset: CGVector(
-                    dx: CGFloat(requestBody.endX),
-                    dy: CGFloat(requestBody.endY)
-            ))
-            
-            logger.info("Swiping from \(startPoint) to \(endPoint) with \(velocity.rawValue) velocity")
-            
-            startPoint.press(
-                forDuration: 0.05,
-                thenDragTo: endPoint,
-                withVelocity: velocity,
-                thenHoldForDuration: 0.0
-            )
-            
-            return HTTPResponse(statusCode: .ok)
-        }
-        
-        return response
+
+
+        try await swipePrivateAPI(
+            start: requestBody.start,
+            end: requestBody.end,
+            duration: requestBody.duration)
+
+        return HTTPResponse(statusCode: .ok)
     }
-    
+
+    func swipePrivateAPI(start: CGPoint, end: CGPoint, duration: Double) async throws {
+
+        logger.info("Swiping from \(start.debugDescription) to \(end.debugDescription) with \(duration) duration")
```

Comment:
nit : I think .debugDescription is not needed, because Swift will use it by default when interpolating string inside log

Reply:
It is was needed because there were multiple implementations for string interpolation and the compiler couldn't decide. However, that may have been for CGVector, which I was using to test both old uielement and private api implementation.

Reply:
<img width="1083" alt="Screenshot 2023-02-14 at 12 46 52" src="https://user-images.githubusercontent.com/915211/218729386-e31eef18-cd9a-4719-a0d7-1450d1ad8941.png">
Meh..

Reply:
hmm, looks like smth os.log specific, works well with normal strings and print() 🤔